### PR TITLE
Bug 1959981: oVirt: fix empty AffinityGroupsNames handling

### PR DESCRIPTION
**Pull request by Gal-Zaidman**
- Handle empty affinity group names field

### CODE REVIEW

File: pkg/types/ovirt/defaults/machinepools.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ func setMachinePool(p *types.MachinePool) {
 }
 
 func setDefaultAffinityGroups(p *ovirt.Platform, mp *types.MachinePool, agName string) {
-	if len(mp.Platform.Ovirt.AffinityGroupsNames) == 0 {
+	if mp.Platform.Ovirt.AffinityGroupsNames == nil {
```
Comment (janoszen) [inline pkg/types/ovirt/defaults/machinepools.go:15]: Do we want this code to run if the AGN is an empty slice? Len seems to be the better option here.

Comment (Gal-Zaidman) [reply pkg/types/ovirt/defaults/machinepools.go:15]: And yes I want to handle an empty slice, meaning:
- affinity_groups_names: [] - disables it
- not writing anything will be the default behavior

Comment (janoszen) [reply pkg/types/ovirt/defaults/machinepools.go:15]: This should be documented since [] is usually equivalent to nil.

Comment (staebler) [reply pkg/types/ovirt/defaults/machinepools.go:15]: See my concerns [here](https://github.com/openshift/installer/pull/4674#discussion_r580295829).

Comment (Gal-Zaidman) [reply pkg/types/ovirt/defaults/machinepools.go:15]: handled by removing the "omitempty" from the field.

Regarding the docs this is the same logic as the platform fields, we will document both in the Docs of course :)

**Pull request merged**